### ISSUE TITLE
[OJS-3.4.0-1] PHP Fatal error:  Uncaught Error: Class "FormValidatorPost" not found in 

### ISSUE THREAD

**Issue by shantanu198713** [RBoelter/citations#14]
OJS-3.4.0.1
PHP-8.2

**Error:** `Failed Ajax request or invalid JSON returned.`
`````
PHP Fatal error:  Uncaught Error: Class "FormValidatorPost" not found in F:\wamp64\www\ojs3401\plugins\generic\citations\CitationsSettingsForm.inc.php:14
Stack trace:
#0 F:\wamp64\www\ojs3401\plugins\generic\citations\CitationsPlugin.inc.php(142): CitationsSettingsForm->__construct(Object(CitationsPlugin))
#1 F:\wamp64\www\ojs3401\lib\pkp\classes\controllers\grid\plugins\PluginGridHandler.php(282): CitationsPlugin->manage(Array, Object(APP\core\Request))
#2 [internal function]: PKP\controllers\grid\plugins\PluginGridHandler->manage(Array, Object(APP\core\Request))
#3 F:\wamp64\www\ojs3401\lib\pkp\classes\core\PKPRouter.php(334): call_user_func(Array, Array, Object(APP\core\Request))
#4 F:\wamp64\www\ojs3401\lib\pkp\classes\core\PKPComponentRouter.php(291): PKP\core\PKPRouter->_authorizeInitializeAndCallRequest(Array, Object(APP\core\Request), Array)
#5 F:\wamp64\www\ojs3401\lib\pkp\classes\core\Dispatcher.php(165): PKP\core\PKPComponentRouter->route(Object(APP\core\Request))
#6 F:\wamp64\www\ojs3401\lib\pkp\classes\core\PKPApplication.php(373): PKP\core\Dispatcher->dispatch(Object(APP\core\Request))
#7 F:\wamp64\www\ojs3401\index.php(21): PKP\core\PKPApplication->execute()
#8 {main}
  thrown in F:\wamp64\www\ojs3401\plugins\generic\citations\CitationsSettingsForm.inc.php on line 14
`````

**Comment by shantanu198713** [comment]
Hi @RBoelter 

I had implemented the change made by you but still, I am facing the same problem.
`Failed Ajax request or invalid JSON returned.`

`````
PHP Fatal error:  Uncaught Error: Class "FormValidatorPost" not found in F:\wamp64\www\ojs3401\plugins\generic\citations\CitationsSettingsForm.inc.php:14
Stack trace:
#0 F:\wamp64\www\ojs3401\plugins\generic\citations\CitationsPlugin.inc.php(152): CitationsSettingsForm->__construct(Object(CitationsPlugin))
#1 F:\wamp64\www\ojs3401\lib\pkp\classes\controllers\grid\plugins\PluginGridHandler.php(282): CitationsPlugin->manage(Array, Object(APP\core\Request))
#2 [internal function]: PKP\controllers\grid\plugins\PluginGridHandler->manage(Array, Object(APP\core\Request))
#3 F:\wamp64\www\ojs3401\lib\pkp\classes\core\PKPRouter.php(334): call_user_func(Array, Array, Object(APP\core\Request))
#4 F:\wamp64\www\ojs3401\lib\pkp\classes\core\PKPComponentRouter.php(291): PKP\core\PKPRouter->_authorizeInitializeAndCallRequest(Array, Object(APP\core\Request), Array)
#5 F:\wamp64\www\ojs3401\lib\pkp\classes\core\Dispatcher.php(165): PKP\core\PKPComponentRouter->route(Object(APP\core\Request))
#6 F:\wamp64\www\ojs3401\lib\pkp\classes\core\PKPApplication.php(373): PKP\core\Dispatcher->dispatch(Object(APP\core\Request))
#7 F:\wamp64\www\ojs3401\index.php(21): PKP\core\PKPApplication->execute()
#8 {main}
  thrown in F:\wamp64\www\ojs3401\plugins\generic\citations\CitationsSettingsForm.inc.php on line 14
`````
`````
PHP Fatal error:  Uncaught Error: Class "JSONMessage" not found in F:\wamp64\www\ojs3401\plugins\generic\citations\classes\CitationsHandler.inc.php:63
Stack trace:
#0 [internal function]: CitationsHandler->get(Array, Object(APP\core\Request))
#1 F:\wamp64\www\ojs3401\lib\pkp\classes\core\PKPRouter.php(334): call_user_func(Array, Array, Object(APP\core\Request))
#2 F:\wamp64\www\ojs3401\lib\pkp\classes\core\PKPPageRouter.php(277): PKP\core\PKPRouter->_authorizeInitializeAndCallRequest(Array, Object(APP\core\Request), Array, false)
#3 F:\wamp64\www\ojs3401\lib\pkp\classes\core\Dispatcher.php(165): PKP\core\PKPPageRouter->route(Object(APP\core\Request))
#4 F:\wamp64\www\ojs3401\lib\pkp\classes\core\PKPApplication.php(373): PKP\core\Dispatcher->dispatch(Object(APP\core\Request))
#5 F:\wamp64\www\ojs3401\index.php(21): PKP\core\PKPApplication->execute()
#6 {main}
  thrown in F:\wamp64\www\ojs3401\plugins\generic\citations\classes\CitationsHandler.inc.php on line 63
`````

**Comment by RBoelter** [comment]
I fixed a bug for 3.3.x.x that seemed similar to yours, but it seems that it doesn't work for OJS 3.4.x. I haven't tested this plugin for OJS 3.4 yet.

**Comment by shantanu198713** [comment]
Thanks for your reply.
I hope soon it will be compatible with ojs-3.4.x.

**Comment by RBoelter** [comment]
It will take a while because I have to rewrite the whole Plugin for 3.4.x 😒

**Comment by RBoelter** [comment]
I have published a beta for 3.4. here: https://github.com/RBoelter/citations/releases/tag/v3_4_0-1
Feel free to test it (no warranty). You need to download the tar.gz file and upload it manually to your OJS via the plugin page or sftp/ssh. I will add it to the OJS plugin library in a few days.

**Comment by shantanu198713** [comment]
Hi @RBoelter 
Now working fine.
